instrumentation/aws-lambda/aws-lambda-events-2.2/library/src/main/java/io/opentelemetry/instrumentation/awslambdaevents/v2_2/internal/SerializationUtil.java

```diff
@@ -0,0 +1,133 @@
+      // For "java8" runtime, "aws-lambda-java-serialization" library
+      // is not available in the classpath by default.
+      // So fall back to object mapper based legacy serialization.
+      return new ObjectMapperPojoSerializer<T>(clazz);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <T> PojoSerializer<T> getSerializer(Class<T> clazz) {
+    return (PojoSerializer<T>) serializerCache.get(clazz);
+  }
+
+  public static <T> T fromJson(String json, Class<T> clazz) {
+    PojoSerializer<T> serializer = getSerializer(clazz);
+    return serializer.fromJson(json);
+  }
+
+  public static <T> T fromJson(InputStream inputStream, Class<T> clazz) {
+    PojoSerializer<T> serializer = getSerializer(clazz);
+    return serializer.fromJson(inputStream);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <T> void toJson(OutputStream outputStream, T obj) {
+    if (obj != null) {
+      PojoSerializer<T> serializer = getSerializer((Class<T>) obj.getClass());
+      serializer.toJson(obj, outputStream);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <T> String toJson(T obj) {
+    if (obj == null) {
+      return null;
+    }
+    PojoSerializer<T> serializer = getSerializer((Class<T>) obj.getClass());
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream(DEFAULT_BUFFER_SIZE);
+    serializer.toJson(obj, outputStream);
+    return new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
+  }
+
+  public static <T> byte[] toJsonData(T obj) {
+    if (obj == null) {
+      return new byte[] {};
+    }
+    ByteArrayOutputStream os = new ByteArrayOutputStream(DEFAULT_BUFFER_SIZE);
+    SerializationUtil.toJson(os, obj);
+    return os.toByteArray();
+  }
+
+  private static class ObjectMapperPojoSerializer<T> implements PojoSerializer<T> {
+
+    private final ObjectMapper objectMapper =
+        new ObjectMapper()
+            .registerModule(new CustomJodaModule())
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private final Class<T> clazz;
+
+    ObjectMapperPojoSerializer(Class<T> clazz) {
+      this.clazz = clazz;
+    }
+
+    @Override
+    public T fromJson(InputStream input) {
+      try {
+        return objectMapper.readValue(input, clazz);
+      } catch (IOException e) {
+        throw new IllegalStateException("Could not deserialize from JSON input stream.", e);
+      }
+    }
+
+    @Override
+    public T fromJson(String input) {
+      try {
+        return objectMapper.readValue(input, clazz);
+      } catch (IOException e) {
+        throw new IllegalStateException("Could not deserialize from JSON string.", e);
+      }
+    }
+
+    @Override
+    public void toJson(T value, OutputStream output) {
+      try {
+        objectMapper.writeValue(output, value);
+      } catch (IOException e) {
+        throw new IllegalStateException("Could not serialize to JSON.", e);
+      }
+    }
+  }
+
+  private SerializationUtil() {}
+}
```

Comment:
I'd like to see new classes like this show up with unit test coverage.

Reply:
Yep, this class is indirectly tested by existing and new tests but not directly. I will add tests to test `SerializationUtil` directly.

Reply:
done